rfcs/0002-branch-generation.md

```diff
@@ -0,0 +1,76 @@
+# Branch Generation
+
+The goal of this RFC is to define how Solana generates branches.
+
+
+## Basic Design Idea
+
+Nodes take turns being leader and generating the PoH that encodes state changes.  The network can tolerate loss of connection to any leader by synthesizing what the leader ***would have generated*** had it been connected but not ingesting any state changes.  The complexity of forks is thereby limited to a "there/not-there" skip list of branches that may arise on leader rotation periods boundaries.  A leader can only transmit durring their predefined PoH slot.
+
+## Message Flow
+
+1. Transactions are ingested at the current leader.
+2. Leader filters for valid transactions.
+3. Leader executes valid transactions on its state.
+4. Leader packages transactions into entries based off its current PoH slot.
+5. Leader transmits the entries to validator nodes (in signed blobs)
+    a. The PoH stream includes ticks; empty entries that indicate liveness of the leader and the passage of time on the network.
+    b. A leader's stream begins with the tick entries necessary complete the PoH back to the leaders most recently observed prior leader period.
+6. Validators retransmit entries to peers in their set and to further downstream nodes.
```

Comment:
"in their set" should either be removed or should be followed up with a description of what defines that set.